.github/workflows/push.yml

```diff
@@ -14,57 +13,102 @@ env:
+  APP_IMAGE: sapi-python-client
 
 jobs:
-  build_and_test:
+  build:
     runs-on: ubuntu-latest
     outputs:
-      is_semantic_tag: ${{steps.tag.outputs.is_semantic_tag}}
+      is_semantic_tag: ${{ steps.tag.outputs.is_semantic_tag }}
     steps:
-      - name: Checkout the repo
+      - name: Check out the repo
         uses: actions/checkout@v3
+
       - name: Print Docker version
         run: |
           docker -v
+
       - name: Docker login
         if: env.DOCKERHUB_TOKEN
         run: docker login --username "$DOCKERHUB_USER" --password "$DOCKERHUB_TOKEN"
-      - name: build
-        run: docker-compose build sapi-python-client
-      - name: show images
-        run: docker images
-      - name: run tests flake8
-        run: docker-compose run --rm --entrypoint=flake8 sapi-python-client
-      - name: run tests aws
-        run: docker-compose run --rm -e KBC_TEST_TOKEN=$KBC_TEST_TOKEN -e KBC_TEST_API_URL=$KBC_TEST_API_URL -e SKIP_ABS_TESTS=1 sapi-python-client -m unittest
-      - name: run tests azure
-        run: docker-compose run --rm -e KBC_TEST_TOKEN=$KBC_AZ_TEST_TOKEN -e KBC_TEST_API_URL=$KBC_AZ_TEST_API_URL -e SKIP_ABS_TESTS=0 sapi-python-client -m unittest
-
-      - name: Check if tag is semantic
+
+      - name: Build image
+        run: docker build -t $APP_IMAGE .
+
+      - name: Check image
+        run: docker-compose run --rm sapi-python-client -m flake8
+
+      - name: Set image tag
         id: tag
         run: |
           TAG="${GITHUB_REF##*/}"
           IS_SEMANTIC_TAG=$(echo "$TAG" | grep -q '^v\?[0-9]\+\.[0-9]\+\.[0-9]\+$' && echo true || echo false)
           echo "Tag = '$TAG', is semantic tag = '$IS_SEMANTIC_TAG'"
           echo "is_semantic_tag=$IS_SEMANTIC_TAG" >> $GITHUB_OUTPUT
+
+      - name: Upload image
+        uses: ishworkh/docker-image-artifact-upload@v1
+        with:
+          image: "sapi-python-client"
+          retention_days: "1"
+
+  tests_aws:
+    name: Run tests (AWS)
+    needs: build
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out the repo
+        uses: actions/checkout@v3
+
+      - name: Download image
+        uses: ishworkh/docker-image-artifact-download@v1
+        with:
+          image: "sapi-python-client"
+
+      - name: Run Tests
+        run: |
+          docker-compose run --rm -e KBC_TEST_TOKEN=$KBC_TEST_TOKEN -e KBC_TEST_API_URL=$KBC_TEST_API_URL -e SKIP_ABS_TESTS=1 sapi-python-client -m unittest
+
+  tests_azure:
+    name: Run tests (Azure)
+    needs: build
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out the repo
+        uses: actions/checkout@v3
+
+      - name: Download image
+        uses: ishworkh/docker-image-artifact-download@v1
+        with:
+          image: "sapi-python-client"
+
+      - name: Run Tests
+        run: |
+          docker-compose run --rm -e KBC_TEST_TOKEN=$KBC_AZ_TEST_TOKEN -e KBC_TEST_API_URL=$KBC_AZ_TEST_API_URL -e SKIP_ABS_TESTS=1 sapi-python-client -m unittest
```

Comment:
Do we really want `SKIP_ABS_TESTS=1` for `tests_azure`?

Maybe we can use a condition? `if: !$SKIP_ABS_TESTS` or stg like that?

Reply:
No we don't want to skip them :)

Not sure how the conditon would work

Reply:
Fixed

Reply:
Ok, now it is really fixed

Reply:
No skipped tests on Azure 

![image](https://github.com/keboola/sapi-python-client/assets/4319320/d8136d77-1b09-4e77-aea5-9dd9d1a7a769)